test/cql-pytest/test_secondary_index.py

```diff
@@ -1269,3 +1269,210 @@ def test_index_paging_group_by(cql, test_keyspace, use_group_by):
             all_rows.extend(results.current_rows)
             # Finally check that altogether, we read the right rows.
             assert sorted(all_rows) == [(i,) for i in range(10)]
+
+# Tests basic operations on a static column index.
+def test_static_column_index(cql, test_keyspace):
+    schema = 'pk int, c int, s int STATIC, v int, PRIMARY KEY(pk, c)'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        cql.execute(f'CREATE INDEX ON {table}(s)')
+
+        # Insert
+        cql.execute(f'INSERT INTO {table} (pk, s) VALUES (0, 0)')
+        cql.execute(f'INSERT INTO {table} (pk, s) VALUES (1, 0)')
+        cql.execute(f'INSERT INTO {table} (pk, s) VALUES (2, 1)')
+
+        assert [(0,),(1,)] == sorted(cql.execute(f'SELECT pk FROM {table} WHERE s = 0'))
+        assert [(2,)] == sorted(cql.execute(f'SELECT pk FROM {table} WHERE s = 1'))
+
+        # Update
+        cql.execute(f'UPDATE {table} SET s = 1 WHERE pk = 1')
+
+        assert [(0,)] == sorted(cql.execute(f'SELECT pk FROM {table} WHERE s = 0'))
+        assert [(1,),(2,)] == sorted(cql.execute(f'SELECT pk FROM {table} WHERE s = 1'))
+
+        # Partition delete
+        cql.execute(f'DELETE FROM {table} WHERE pk = 2')
+
+        assert [(0,)] == sorted(cql.execute(f'SELECT pk FROM {table} WHERE s = 0'))
+        assert [(1,)] == sorted(cql.execute(f'SELECT pk FROM {table} WHERE s = 1'))
+
+# Tests that building static indexes from a non-empty state works.
+def test_static_column_index_build(cql, test_keyspace):
+    schema = 'pk int, c int, s int STATIC, v int, PRIMARY KEY(pk, c)'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        cql.execute(f'INSERT INTO {table} (pk, s) VALUES (0, 0)')
+        cql.execute(f'INSERT INTO {table} (pk, s) VALUES (1, 0)')
+        cql.execute(f'INSERT INTO {table} (pk, s) VALUES (2, 0)')
+        cql.execute(f'CREATE INDEX ON {table}(s)')
+
+        # Indexes are created in the background, so we should wait here.
+        # I don't know how to get information about secondary index build
+        # status on C*, so we'll just wait until 30 seconds elapse or
+        # the index appears to be properly built.
+        start_time = time.time()
+        rows = None
+        while time.time() < start_time + 5:
```

Comment:
The comment says 30 seconds, but the code says 5 seconds. Let's increase it to 30 seconds, I'd hate for us to run into this timeout in one of the super-slow test builds.
In any case, it's just a worst-case timeout, you don't usually need to wait that long.